File: Basic Level/basicLevel1030/Main.java
package basicLevel1030;
//测试点4超时
import java.util.Arrays;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		int p = in.nextInt();
		
		int[] array = new int[n];
		for (int i = 0; i < n; i++) {
			array[i] = in.nextInt();
		}
		in.close();
		
		Arrays.sort(array);
		
		int cnt = 1;
	    for (int i = n-1; i >= 1; i--) {
	        int temp = 1;
	        int M = array[i];
	        for (int j = i-cnt; j >= 0 && M * 1.0 / array[j] <= p ; j--) {
	            temp = i - j + 1;
	        }
	        cnt = cnt > temp ? cnt : temp;
	    }
		
		System.out.println(cnt);
	}
}

File: Basic Level/basicLevel1030/main.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    int n = 0, p = 0;
    cin >> n >> p;
    long long a[100001];    
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    
    if (n == 0 || p < 0) {
        cout << n;
        return 0;
    }
    
    sort(a, a + n);
    
    long cnt = 1;
    for (int i = n-1; i >= 1; i--) {
        int temp = 1;
        long long M = a[i];
        for (int j = i-cnt; j >= 0 && M * 1.0 / a[j] <= p ; j--) {
            temp = i - j + 1;
        }
        cnt = cnt > temp ? cnt : temp;
    }
    
    cout << cnt;
    return 0;
}